config/consensus.go

```diff
@@ -351,6 +351,15 @@ type ConsensusParams struct {
 
 	// NoEmptyLocalDeltas updates how ApplyDelta.EvalDelta.LocalDeltas are stored
 	NoEmptyLocalDeltas bool
+
+	// EnableKeyregCoherencyCheck enable the following extra checks on key registration transactions:
+	// 1. checking that [VotePK/SelectionPK/VoteKeyDilution] are all set or all clear.
+	// 2. checking that the VoteFirst is less or equal to VoteLast.
+	// 3. checking that in the case of going offline, both the VoteFirst and VoteLast are clear.
+	// 4. checking that in the case of going online the VoteLast is non-zero and greater then the current network round.
+	// 5. checking that in the case of going online the VoteFirst is less or equal to the LastValid+1.
+	// 6. checking that in the case of going online the VoteFirst is less or equal to the <voting round>+1.
```

Comment:
What is `<voting round>`?

Reply:
would probably be a good idea to clarify this further. in here, `<voting round>` is the round in which we attempt to apply the said transaction. ( i.e. the voting using the participation key itself can happen not before the round is accepted )